packages/astro/src/vite-plugin-astro/index.ts

```diff
@@ -103,35 +102,22 @@ export default function astro({ settings, logging }: AstroPluginOptions): vite.P
 			if (!query.astro) {
 				return null;
 			}
-			let filename = parsedId.filename;
-			// For CSS / hoisted scripts we need to load the source ourselves.
-			// It should be in the compilation cache at this point.
-			let raw = await this.resolve(filename, undefined);
-			if (!raw) {
-				return null;
-			}
-
-			let source = getCachedSource(config, raw.id);
-			if (!source) {
+			// For CSS / hoisted scripts, the main Astro module should already be cached
+			const filename = normalizeFilename(parsedId.filename, config);
+			const compileResult = getCachedCompileResult(config, filename);
+			if (!compileResult) {
```

Comment:
I'm not sure this is right. At least looking at the old code we used to do `await cachedCompilation(compileProps);` for styles. Although thinking about it now, there shouldn't be a time where styles are loaded before we have compiled 🤔

Reply:
I noticed that for styles too but we always [guarded this part](https://github.com/withastro/astro/blob/3c44033e4ebafd28472d5c6a43e55c0363c6b555/packages/astro/src/vite-plugin-astro/index.ts#L114-L117) with a:

```js
let source = getCachedSource(config, raw.id);
if (!source) {
	return null;
}
```

So technically we're expecting the main module to be compiled before we process the styles, otherwise we bail.

Reply:
I would verify that all of the tests pass with this.